After installing MongoDb and attempting transfer from neb database, all devices & user gone and it appears to be a blank slate.

**Describe your issue**
After installing MongoDb and attempting to transfer neb database to MongoDB, all devices & users gone and it appears to be a blank slate.  Looks like a fresh install.

Trying to run node node_modules/meshcentral --nedbtodb and get the error below.

I tried opening up meshcentral.db (only to view not save) and all my info is there in that file just not in new MongoDb Database.

**Screenshots**

```
Transfering NeDB data into database...
ERR: internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'nedb'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.obj.nedbtodb (/opt/mc/node_modules/meshcentral/db.js:3136:29)
    at /opt/mc/node_modules/meshcentral/meshcentral.js:963:28
    at /opt/mc/node_modules/meshcentral/db.js:102:13
[servererrors.txt](https://github.com/Ylianst/MeshCentral/files/9485321/servererrors.txt)

    at /opt/mc/node_modules/meshcentral/db.js:2172:29
    at wrapper (/opt/mc/node_modules/mongodb/lib/utils.js:530:9)
    at next (/opt/mc/node_modules/mongodb/lib/cursor/abstract_cursor.js:244:32)
{ Error: Command failed: /usr/bin/node /opt/mc/node_modules/meshcentral --nedbtodb --launch 4049
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'nedb'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.obj.nedbtodb (/opt/mc/node_modules/meshcentral/db.js:3136:29)
    at /opt/mc/node_modules/meshcentral/meshcentral.js:963:28
    at /opt/mc/node_modules/meshcentral/db.js:102:13
    at /opt/mc/node_modules/meshcentral/db.js:2172:29
    at wrapper (/opt/mc/node_modules/mongodb/lib/utils.js:530:9)
    at next (/opt/mc/node_modules/mongodb/lib/cursor/abstract_cursor.js:244:32)

    at ChildProcess.exithandler (child_process.js:294:12)
    at ChildProcess.emit (events.js:203:15)
    at maybeClose (internal/child_process.js:982:16)
    at Socket.stream.socket.on (internal/child_process.js:389:11)
    at Socket.emit (events.js:198:13)
    at Pipe._handle.close (net.js:607:12)
  killed: false,
  code: 1,
  signal: null,
  cmd:
   '/usr/bin/node /opt/mc/node_modules/meshcentral --nedbtodb --launch 4049' }
ERROR: MeshCentral failed with critical error, check mesherrors.txt. Restarting in 5 seconds...
```

**Server Software (please complete the following information):**
 - OS: [Ubuntu]
 - Virtualization: [no
 - Network: [WAN]
 - Version: [1.0.70]
 - Node: [10.19.0]

**Client Device (please complete the following information):**
 - Device: [e.g. Laptop]
 - OS: [e.g. Ubuntu]
 - Network: [e.g. Local to Meshcentral, Remote over WAN]
 - Browser: [e.g. Google Chrome]
 - MeshCentralRouter Version: [if applicable]

**Remote Device (please complete the following information):**
 - Device: [e.g. Laptop]
 - OS: [e.g. Windows 10 21H2]
 - Network: [e.g. Local to Meshcentral, Remote over WAN]
 - Current Core Version (if known): [**HINT**: Go to a device then `console` Tab then type `info`]

**Your config.json file**
```
{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "mycert.net",
    "WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "port": 443,
    "MongoDb": "mongodb://127.0.0.1:27017",
    "_aliasPort": 443,
    "redirPort": 80,
    "_redirAliasPort": 80
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "left out",
    "names": "my server.net",
    "rsaKeySize": 3072,
    "production": true
  }
}
                                            
```

--- a/db.js
+++ b/db.js
@@ -1470,7 +1470,14 @@ module.exports.CreateDB = function (parent, func) {
             obj.StoreEvent = function (event, func) {
                 obj.dbCounters.eventsSet++;
                 sqlDbQuery('INSERT INTO events VALUES (NULL, $1, $2, $3, $4, $5, $6) RETURNING id', [event.time, ((typeof event.domain == 'string') ? event.domain : null), event.action, event.nodeid ? event.nodeid : null, event.userid ? event.userid : null, JSON.stringify(event)], function (err, docs) {
-                    if ((err == null) && (docs[0].id)) { for (var i in event.ids) { if (event.ids[i] != '*') { sqlDbQuery('INSERT INTO eventids VALUES ($1, $2)', [docs[0].id, event.ids[i]]); } } }
+                    if(func){ func(); }
+                    if ((err == null) && (docs[0].id)) {
+                        for (var i in event.ids) {
+                            if (event.ids[i] != '*') {
+                                sqlDbQuery('INSERT INTO eventids VALUES ($1, $2)', [docs[0].id, event.ids[i]], function(){ if(func){ func(); } });
+                            }
+                        }
+                    }
                 });
             };
             obj.GetEvents = function (ids, domain, func) {
@@ -3239,6 +3246,7 @@ module.exports.CreateDB = function (parent, func) {
                     for (var i in docs) {
                         pendingTransfer++;
                         eventRecordsTransferCount++;
+                        for (var b in docs[i].ids) { if (docs[i].ids[b] != '*') { pendingTransfer++; } }
                         obj.StoreEvent(docs[i], function () { pendingTransfer--; });
                     }
                 }

--- a/db.js
+++ b/db.js
@@ -3181,7 +3181,10 @@ module.exports.CreateDB = function (parent, func) {
 
     // Transfer NeDB data into the current database
     obj.nedbtodb = function (func) {
-        var nedbDatastore = require('nedb');
+        var nedbDatastore = null;
+        try { nedbDatastore = require('@yetzt/nedb'); } catch (ex) { } // This is the NeDB with fixed security dependencies.
+        if (nedbDatastore == null) { nedbDatastore = require('nedb'); } // So not to break any existing installations, if the old NeDB is present, use it.
+
         var datastoreOptions = { filename: parent.getConfigFilePath('meshcentral.db'), autoload: true };
 
         // If a DB encryption key is provided, perform database encryption

